tests/test_model.py

```diff
@@ -522,3 +522,417 @@ def test_model_diskonlyfilemodel_field_is_constructed_correctly(
+
+        with open(temporary_save_location, "r") as file:
+            assert any(
+                line.strip() == line_to_check for line in file
+            ), f"File {temporary_save_location} does not contain the line: {line_to_check}"
+
+    @pytest.mark.parametrize(
+        "input_process",
+        [
+            pytest.param("0 1"),
+            pytest.param("0 1 2"),
+            pytest.param("0 1 2 3"),
+            pytest.param("0 1 2 3 4"),
+        ],
+    )
+    def test_dimr_with_fmcomponent_given_correct_style_for_setting_process(
+        self, tmp_path, input_process
+    ):
+        dimr_config_data = self.get_fm_dimr_config_data(input_process)
+        (
+            temporary_dimr_config_file,
+            temporary_save_location,
+        ) = self.setup_temporary_files(tmp_path, dimr_config_data)
+
+        dimr_config = DIMR(filepath=temporary_dimr_config_file)
+        dimr_config.save(filepath=temporary_save_location)
+
+        assert_files_equal(temporary_dimr_config_file, temporary_save_location)
+
+    def test_dimr_with_fmcomponent_given_correct_style_for_setting_process_for_zero(
+        self, tmp_path
+    ):
+        dimr_config_data = self.get_fm_dimr_config_data("0")
+
+        (
+            temporary_dimr_config_file,
+            temporary_save_location,
+        ) = self.setup_temporary_files(tmp_path, dimr_config_data)
+
+        dimr_config_data_expected = self.get_fm_dimr_config_data_without_process()
+        temporary_expected_dimr_config_file = tmp_path / "dimr_expected_config.xml"
+        temporary_expected_dimr_config_file.write_text(dimr_config_data_expected)
+
+        dimr_config = DIMR(filepath=temporary_dimr_config_file)
+        dimr_config.save(filepath=temporary_save_location)
+
+        assert_files_equal(temporary_expected_dimr_config_file, temporary_save_location)
+
+    @pytest.mark.parametrize(
+        "input_process, expected_process",
+        [
+            pytest.param("0", 1),
+            pytest.param("0 1", 2),
+            pytest.param("0 1 2", 3),
+            pytest.param("0 1 2 3", 4),
+            pytest.param("0 1 2 3 4", 5),
+            pytest.param("0:1", 2),
+            pytest.param("0:2", 3),
+            pytest.param("0:3", 4),
+            pytest.param("0:4", 5),
+            pytest.param("2:4", 3),
+        ],
+    )
+    def test_dimr_with_fmcomponent_process_component_set_correctly(
+        self, tmp_path, input_process, expected_process
+    ):
+        dimr_config_data = self.get_fm_dimr_config_data(input_process)
+        (
+            temporary_dimr_config_file,
+            temporary_save_location,
+        ) = self.setup_temporary_files(tmp_path, dimr_config_data)
+
+        dimr_config = DIMR(filepath=temporary_dimr_config_file)
+        dimr_config.save(filepath=temporary_save_location)
+
+        assert dimr_config.component[0].process == expected_process
+
+    def test_dimr_with_fmcomponent_given_old_incorrect_style_for_setting_process(
```

Comment:
Please add to the name that old style should raise error so we dont have to look at the test to see what the expected behaviour is,

Reply:
added _raise_valueerror